### PR TITLE
SPU2: Remove Waveout backend from Windows

### PR DESCRIPTION
**Pull request by Mrlinkwii**
### Description of Changes

Remove Waveout backend from Windows
### Rationale behind Changes

It's borked  
### Suggested Testing Steps

Make sure GUI on windows is OK and other modules still work

### CODE REVIEW

**Comment by RedDevilus** [comment]
This is the story of WaveOut, a devilish spawn to induce nightmares and wreak havoc upon the ignorant sheeplings:


https://user-images.githubusercontent.com/24227051/128613696-531798a2-e2a8-477e-a943-a55a1901aa18.mp4


Steps to reproduce:

1. Open PCSX2 (1.7)
2. Go to Audio Settings > Module > WaveOut > OK
3. Boot any game or BIOS with full/fast boot
4. Should hear above video 
image

----------
5. Press ESC or pause the game in the GUI
6. Press resume > Should hear better audio
----------
7. Reset TimeStretch settings
image
8. Game will break again with the sound if you boot


Notes any action to pause or fully close PCSX2 will fix this behavior however any action to change to module WaveOut or any settings touching upon it will make it sound like a nightmare.

**Comment by tadanokojin** [comment]
Let's get the purge master on this to make sure you guys didn't miss anything.

File: pcsx2/SPU2/Windows/Config.cpp [hunk 1]
```diff
@@ -127,9 +126,7 @@ void ReadSettings()
 	dspPluginModule = CfgReadInt(L"DSP PLUGIN", L"ModuleNum", 0);
 	dspPluginEnabled = CfgReadBool(L"DSP PLUGIN", L"Enabled", false);
 
-	// Read WAVEOUT configs:
-	CfgReadStr(L"WAVEOUT", L"Device", Config_WaveOut.Device, L"default");
-	Config_WaveOut.NumBuffers = CfgReadInt(L"WAVEOUT", L"Buffer_Count", 4);
+
```
Comment (lightningterror) [inline pcsx2/SPU2/Windows/Config.cpp:130]: Just a minor thing, remove the 2 empty lines here.

File: pcsx2/SPU2/Windows/Config.cpp [hunk 2]
```diff
@@ -192,8 +184,7 @@ void CheckOutputModule(HWND window)
 {
 	OutputModule = SendMessage(GetDlgItem(window, IDC_OUTPUT), CB_GETCURSEL, 0, 0);
 	const bool IsConfigurable =
-		mods[OutputModule] == PortaudioOut ||
-		mods[OutputModule] == WaveOut;
+		mods[OutputModule] == PortaudioOut;
```
Comment (lightningterror) [inline pcsx2/SPU2/Windows/Config.cpp:187]: You can move this on the same row as IsConfigurable

Comment (Mrlinkwii) [reply pcsx2/SPU2/Windows/Config.cpp]: done :)

File: pcsx2/SPU2/Windows/Config.cpp (hunk 1 above)
Comment (Mrlinkwii) [reply pcsx2/SPU2/Windows/Config.cpp]: done :)

**Review by tadanokojin — APPROVED** [review]
I've got no problem removing it.

**Pull request merged**